packages/ember-application/tests/system/visit_test.js

```diff
@@ -35,6 +37,67 @@ moduleFor('Application - visit()', class extends ApplicationTestCase {
     );
   }
 
+  [`@test does not add serialize-mode markers by default`](assert) {
+    let templateContent = '<div class="foo">Hi, Mom!</div>';
+    this.addTemplate('index', templateContent);
+    let rootElement = document.createElement('div');
+
+    let bootOptions = {
+      isBrowser: false,
+      rootElement
+    };
+
+    ENV._APPLICATION_TEMPLATE_WRAPPER = false;
+    return this.visit('/', bootOptions).then(()=> {
+      assert.equal(rootElement.innerHTML, templateContent, 'without serialize flag renders as expected');
+    });
+  }
+
+  [`@test _renderMode: rehydration`](assert) {
+    assert.expect(2);
+
+    let indexTemplate = '<div class="foo">Hi, Mom!</div>';
+    this.addTemplate('index', indexTemplate);
+    let rootElement = document.createElement('div');
+
+    let bootOptions = {
+      isBrowser: false,
+      rootElement,
+      _renderMode: 'serialize'
+    };
+
+    ENV._APPLICATION_TEMPLATE_WRAPPER = false;
+
+    return this.runTask(() => {
+      return this.visit('/', bootOptions)
+        .then((instance) => {
+          assert.equal(
+            instance.rootElement.firstChild.nodeValue,
+            '%+b:0%',
```

Comment:
I don't really like the coupling to the internal / private mechanism that  Glimmer's serialize mode renderer uses internally (generally speaking I'd much prefer to test the _results_ rather than _how_ it works internally) but I realize this is somewhat of a chicken or the egg situation. 

Thoughts?

Reply:
I totally agree that I'd prefer not to couple so tightly to the internal implementation.  Though to ensure we are actually doing some work this seems to be necessary.

I'd like in the future to expose the leading comment node in the serializer builder in glimmer as an export so we don't have to hardcode this value.  Perhaps that could be a next step?